### PR TITLE
GitHub Actions

### LINKED ISSUE #14: Build consistently timing out with Julia nightly on Linux
It looks like this started happening recently:

* https://travis-ci.org/JuliaPackaging/CMakeWrapper.jl/jobs/410847838
* https://travis-ci.org/JuliaPackaging/CMakeWrapper.jl/jobs/410847842

But I'm also seeing the same timeout in downstream dependencies, so whatever is going on is also affecting the most recent release of CMakeWrapper (which predates splitting the package off into CMake.jl).  

* https://travis-ci.org/JuliaRobotics/LCMCore.jl/jobs/410548179

### LINKED ISSUE #15: Build fails on Julia 1.0
Apologies if this is not appropriate due to future plans of splitting CMakeWrapper into 2 packages, but trying to to build this on Windows 10 Pro 64 bit system I get the following error:
```

(v1.0) pkg> build "CMakeWrapper"
  Building CMakeWrapper → `C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps\build.log`
┌ Error: Error building `CMakeWrapper`:
│ [ Info: Attempting to create directory C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps\downloads
│ [ Info: Directory C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps\downloads already exists
│ [ Info: Downloading file https://cmake.org/files/v3.7/cmake-3.7.2-win64-x64.zip
│ [ Info: Done downloading file https://cmake.org/files/v3.7/cmake-3.7.2-win64-x64.zip
│ [ Info: Attempting to create directory C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps
│ [ Info: Directory C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps already exists
│ [ Info: Path C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps\downloads\cmake-3.7.2-win64-x64 already exists
│ [ Info: Attempting to create directory C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps\usr\bin
│ [ Info: Directory C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps\usr\bin already exists
│ ERROR: LoadError: MethodError: no method matching cp(::String, ::String; remove_destination=true)
│ Closest candidates are:
│   cp(::AbstractString, ::AbstractString; force, follow_symlinks) at file.jl:330 got unsupported keyword argument "remove_destination"
│ Stacktrace:
│  [1] kwerr(::NamedTuple{(:remove_destination,),Tuple{Bool}}, ::Function, ::String, ::String) at .\error.jl:97
│  [2] (::getfield(Base.Filesystem, Symbol("#kw##cp")))(::NamedTuple{(:remove_destination,),Tuple{Bool}}, ::typeof(cp), ::String, ::String) at .\none:0
│  [3] (::getfield(Main, Symbol("##3#4")){String})() at C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps\build.jl:23
│  [4] run(::getfield(Main, Symbol("##3#4")){String}) at C:\Users\jonat\.julia\packages\BinDeps\Z6fwm\src\BinDeps.jl:483
│  [5] macro expansion at .\logging.jl:308 [inlined]
│  [6] run(::BinDeps.SynchronousStepCollection) at C:\Users\jonat\.julia\packages\BinDeps\Z6fwm\src\BinDeps.jl:523
│  [7] run(::FileRule) at C:\Users\jonat\.julia\packages\BinDeps\Z6fwm\src\BinDeps.jl:488
│  [8] macro expansion at .\logging.jl:308 [inlined]
│  [9] run(::BinDeps.SynchronousStepCollection) at C:\Users\jonat\.julia\packages\BinDeps\Z6fwm\src\BinDeps.jl:523
│  [10] top-level scope at none:0
│  [11] include at .\boot.jl:317 [inlined]
│  [12] include_relative(::Module, ::String) at .\loading.jl:1038
│  [13] include(::Module, ::String) at .\sysimg.jl:29
│  [14] include(::String) at .\client.jl:388
│  [15] top-level scope at none:0
│ in expression starting at C:\Users\jonat\.julia\packages\CMakeWrapper\R7ZgO\deps\build.jl:124
└ @ Pkg.Operations C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.0\Pkg\src\Operations.jl:1068
```

### LINKED ISSUE #22: precompilation problems
https://ci.appveyor.com/project/musm/hdf5-jl/builds/27542185

For some reason Cmakewrapper keeps failing to precompile?

### PR DESCRIPTION
**Pull request by aminya**
- Fixing `compat` bounds which caused windows to fail to build
- Removed Manifest
- Using GitHub actions instead of Travis and Appveyor
- Testing build from source

Fixes https://github.com/JuliaPackaging/CMakeWrapper.jl/issues/15
Fixes https://github.com/JuliaPackaging/CMakeWrapper.jl/issues/14
Fixes https://github.com/JuliaPackaging/CMakeWrapper.jl/issues/22

To make CodeCov working, you should add the `CODECOV_TOKEN` to the secrets:
- Copy your token from here  https://codecov.io/gh/JuliaPackaging/CMakeWrapper.jl/settings
- Add it here with the name `CODECOV_TOKEN`: https://github.com/JuliaPackaging/CMakeWrapper.jl/settings/secrets

**Notice**: The Appveyor app should be removed from the repository, otherwise it will throw errors because of missing config: 
https://github.com/JuliaPackaging/CMakeWrapper.jl/settings/installations

## Ready

### CODE REVIEW

**Comment by aminya** [comment]
Any feedback on this? This is ready.
@rdeits @ihnorton @giordano

File: Project.toml [hunk 1]
```diff
@@ -11,8 +11,7 @@ Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
-BinDeps = "≥ 0.4.0"
-CMake = "1"
-Parameters = "≥ 0.5.0"
-julia = "≥ 0.7.0"
-
+BinDeps = "0.4, 0.8, 1"
+CMake = "1, 1.2"
```
Comment (rdeits) [inline Project.toml]: I think `CMake = "1"` is still the right thing to do--that will allow any version which is compatible with CMake 1.0 according to semver, meaning any version 1.x.

File: Project.toml [hunk 2]
```diff
@@ -11,8 +11,7 @@ Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
...
+Parameters = "0.5, 0.12"
```
Comment (rdeits) [inline Project.toml]: This needs to be a list of compatible versions (e.g. `"0.5, 0.6, 0.7, 0.8, 0.9, 0.10, 0.11, 0.12"`). As written, it only matches v0.5 or v0.12.

File: README.md [hunk 3]
```diff
@@ -1,7 +1,6 @@
 # CMakeWrapper
 
-[![Build Status](https://travis-ci.org/JuliaPackaging/CMakeWrapper.jl.svg?branch=master)](https://travis-ci.org/JuliaPackaging/CMakeWrapper.jl)
-[![Build status](https://ci.appveyor.com/api/projects/status/x0k8dy47kau99bhr?svg=true)](https://ci.appveyor.com/project/StevenGJohnson/cmakewrapper-jl)
+![CI](https://github.com/aminya/CMakeWrapper.jl/workflows/CI/badge.svg)
```
Comment (rdeits) [inline README.md]: Can you change the URL back to `JuliaPackaging` here?

**Review by rdeits — COMMENTED** [review]
Thanks for the PR, and sorry for the delay! I think switching to github CI looks good. I'll take care of the codecov token and disabling appveyor. 

Can you comment on what the root cause of the Windows compat problem was?

File: Project.toml (hunk 2 above)
Comment (aminya) [reply Project.toml]: Fixed in https://github.com/JuliaPackaging/CMakeWrapper.jl/pull/24/commits/ffb8bf179cdb2442b10b89de3f1ac4bfe2677b20

File: Project.toml (hunk 1 above)
Comment (aminya) [reply Project.toml]: Fixed in https://github.com/JuliaPackaging/CMakeWrapper.jl/pull/24/commits/ffb8bf179cdb2442b10b89de3f1ac4bfe2677b20

File: README.md (hunk 3 above)
Comment (aminya) [reply README.md]: Fixed in https://github.com/JuliaPackaging/CMakeWrapper.jl/pull/24/commits/ee083e81427fa416c55e893621122e08bb963572

**Comment by aminya** [comment]
I addressed the comments. @rdeits

**Review by rdeits — APPROVED** [review]
Looks good, thank you!

**Comment by rdeits** [comment]
I'm merging this now, but it looks like I don't have permissions to set the codecov secrets or disable appveyor. Can one of the repo owners take care of that?

**Pull request merged**